src/Npgsql/NpgsqlCommand.cs

```diff
@@ -383,6 +383,60 @@ public new NpgsqlParameter CreateParameter()
 
         #endregion
 
+        internal void DeriveParametersForQuery()
+        {
+            var connector = CheckReadyAndGetConnector();
+            using (connector.StartUserAction())
+            {
+                Log.Debug($"Deriving Parameters for query: {CommandText}", connector.Id);
+                ProcessRawQuery(true);
```

Comment:
Since NpgsqlCommand.SendParseDescribe() sends an empty string (unnamed prepared statement) in the ParseMessage unless the prepare parameter is set to true, there shouldn't be any interference between explicitly and implicitly prepared statements and the parameter derivation.